client/me/account/main.jsx

```diff
@@ -98,29 +106,28 @@ const INTERFACE_FIELDS = [
 const Account = createReactClass( {
 	displayName: 'Account',
 
-	mixins: [ observe( 'userSettings' ) ],
-
-	propTypes: {
-		userSettings: PropTypes.object.isRequired,
-	},
-
 	UNSAFE_componentWillMount() {
 		// Clear any username changes that were previously made
 		this.clearUsernameValidation();
-		this.props.userSettings.removeUnsavedSetting( 'user_login' );
+		this.props.removeUnsavedUserSetting( 'user_login' );
 	},
 
 	componentDidMount() {
 		debug( this.constructor.displayName + ' component is mounted.' );
-		this.props.userSettings.getSettings();
 		this.debouncedUsernameValidate = debounce( this.validateUsername, 600 );
 	},
 
+	componentDidUpdate() {
+		if ( ! this.hasUnsavedUserSettings( [].concat( ACCOUNT_FIELDS, INTERFACE_FIELDS ) ) ) {
+			this.props.markSaved?.();
```

Comment:
I find it odd that in some instances we call `markSaved()` with optional chaining, and in some, we call it directly. If `markSaved` may not exist under some circumstances, shouldn't we update it to reflect that everywhere in the codebase?

Reply:
You're right. We should assume it's there. I'll remove optional chaining for these methods.

Reply:
Addressed via https://github.com/Automattic/wp-calypso/pull/49760/commits/e9d1984639f9c7ae10c868215e4c5d3ac90af28b